auth: Buildbot compiler error on Ubuntu noble

- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [GitHub Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

 - Program: Authoritative
 - Issue type: Bug report

### Short description
https://builder.powerdns.com/#/builders/99 has been failing since https://builder.powerdns.com/#/builders/99/builds/3411 on 2025-02-14 with a compiler error in `webserver.cc` on Ubuntu noble. Notably that build includes #15114.

### Other information
Excerpt from https://builder.powerdns.com/#/builders/99/builds/3418/steps/20/logs/stdio:

```
#39 821.0 g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp  -I/usr/include   -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1   -I/usr/include/luajit-2.1    -Wdate-time -D_FORTIFY_SOURCE=3 -DSYSCONFDIR=\"/etc/powerdns\" -DPKGLIBDIR=\"/usr/lib/x86_64-linux-gnu/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/pdns/pdns-5.0.0-alpha0.2714.master.g559f5c204=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/pdns/pdns-5.0.0-alpha0.2714.master.g559f5c204=/usr/src/pdns-5.0.0~alpha0+master.2714.g559f5c204-1pdns.ubuntu24.04 -c -o epollmplexer.o epollmplexer.cc
#39 821.4 webserver.cc:467:8: error: 'bitset' in namespace 'std' does not name a template type
#39 821.4   467 |   std::bitset<127> set;
#39 821.4       |        ^~~~~~
#39 821.4 webserver.cc:39:1: note: 'std::bitset' is defined in header '<bitset>'; did you forget to '#include <bitset>'?
#39 821.4    38 | #include <yahttp/router.hpp>
#39 821.4   +++ |+#include <bitset>
#39 821.4    39 | #include <algorithm>
#39 821.4 webserver.cc: In constructor 'ValidChars::ValidChars()':
#39 821.4 webserver.cc:464:10: error: expected unqualified-id before '.' token
#39 821.4   464 |       set.set(letter);
#39 821.4       |          ^
#39 821.4 webserver.cc:463:15: warning: unused variable 'letter' [-Wunused-variable]
#39 821.4   463 |     for (auto letter : "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~:/?#[]@!$&'()*+,;=") {
#39 821.4       |               ^~~~~~
#39 821.4 webserver.cc: In function 'bool validURLChars(const std::string&)':
#39 821.4 webserver.cc:485:55: error: 'const struct ValidChars' has no member named 'set'
#39 821.4   485 |     else if (static_cast<size_t>(*iter) >= validChars.set.size() || !validChars.set[*iter]) {
#39 821.4       |                                                       ^~~
#39 821.4 webserver.cc:485:81: error: 'const struct ValidChars' has no member named 'set'
#39 821.4   485 |     else if (static_cast<size_t>(*iter) >= validChars.set.size() || !validChars.set[*iter]) {
#39 821.4       |                                                                                 ^~~
#39 822.8 make[4]: *** [Makefile:3356: webserver.o] Error 1
```

--- a/pdns/misc.cc
+++ b/pdns/misc.cc
@@ -60,6 +60,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <climits>
+#include <unordered_map>
 #ifdef __FreeBSD__
 #  include <pthread_np.h>
 #endif

--- a/pdns/webserver.cc
+++ b/pdns/webserver.cc
@@ -37,7 +37,7 @@
 #include "uuid-utils.hh"
 #include <yahttp/router.hpp>
 #include <algorithm>
-#include <unordered_set>
+#include <bitset>
 
 json11::Json HttpRequest::json()
 {

